src/L.Control.Heightgraph.js

```diff
@@ -106,20 +106,17 @@ import {
                     .remove();
             }
 
+            this._removeMarkedSegmentsOnMap();
+            this._resetDrag(false);
+
             this._data = data;
+            this._init_options();
```

Comment:
I removed the helper (which I introduced in a previous commit) and added this call to `_init_options()` back where it was. No more mapping issues.